src/merkletree.ts

```diff
@@ -0,0 +1,174 @@
+import { BytesLike, HexString, compare } from './bytes';
+
+import {
+  MultiProof,
+  makeMerkleTree,
+  isValidMerkleTree,
+  getProof,
+  getMultiProof,
+  processProof,
+  processMultiProof,
+  renderMerkleTree,
+} from './core';
+
+import { MerkleTreeOptions, defaultOptions } from './options';
+import { validateArgument, invariant } from './utils/errors';
+
+export type MerkleTreeData<T extends Exclude<any, number>> = {
```

Comment:
`Exclude<any, number>` is the same as `any`. I can see that you're doing this to avoid issues with `getProof`. It does not achieve what you want to achieve.

I think you should use `any` and keep the runtime check that you have in the constructor. There are ways to do it statically but they are complex and I don't think it's worth it.

Reply:
I recall testing it and I'm realizing I got confused because `number` is not compatible with `BytesLike`. Thanks for taking a look.

I see they were removed so solving